lineman/app/components/dashboard_page/dashboard_page.haml

```diff
@@ -62,35 +62,13 @@
             %span{translate: 'dashboard_page.no_threads'}>
             %a{ng-show: 'dashboardPage.filter() != "show_all"', translate: 'dashboard_page.view_recent', href: '#', ng-click: 'dashboardPage.changePreferences({dashboard_filter: "show_all"})'}>
           .dashboard-page-sort-group{ng-if: 'dashboardPage.sort() == "sort_by_date"'}
-            %section.dashboard-page-threads-container{ng-if: 'dashboardPage.anyToday()', role: 'region', aria-label: "{{ 'dashboard_page.threads_from.today' | translate }}"}
-              .dashboard-page-date-range{translate: 'common.date_range.today', aria-hidden: 'true'}
-              .dashboard-page-threads
-                %dashboard_page_thread{discussion: 'discussion', ng-repeat: 'discussion in dashboardPage.dashboardDiscussions() | filter:dashboardPage.today'}
-            %section.dashboard-page-threads-container{ng-if: 'dashboardPage.anyYesterday()', role: 'region', aria-label: "{{ 'dashboard_page.threads_from.yesterday' | translate }}"}
-              .dashboard-page-date-range{translate: 'common.date_range.yesterday', aria-hidden: 'true'}
-              .dashboard-page-threads
-                %dashboard_page_thread{discussion: 'discussion', ng-repeat: 'discussion in dashboardPage.dashboardDiscussions() | filter:dashboardPage.yesterday'}
-            %section.dashboard-page-threads-container{ng-if: 'dashboardPage.anyThisWeek()', role: 'region', aria-label: "{{ 'dashboard_page.threads_from.this_week' | translate }}"}
-              .dashboard-page-date-range{translate: 'common.date_range.this_week', aria-hidden: 'true'}
-              .dashboard-page-threads
-                %dashboard_page_thread{discussion: 'discussion', ng-repeat: 'discussion in dashboardPage.dashboardDiscussions() | filter:dashboardPage.thisWeek'}
-            %section.dashboard-page-threads-container{ng-if: 'dashboardPage.anyThisMonth()', role: 'region', aria-label: "{{ 'dashboard_page.threads_from.this_month' | translate }}"}
-              .dashboard-page-date-range{translate: 'common.date_range.this_month', aria-hidden: 'true'}
-              .dashboard-page-threads
-                %dashboard_page_thread{discussion: 'discussion', ng-repeat: 'discussion in dashboardPage.dashboardDiscussions() | filter:dashboardPage.thisMonth'}
-            %section.dashboard-page-threads-container{ng-if: 'anyOlder()', role: 'region', aria-label: "{{ 'dashboard_page.threads_from.older' | translate }}"}
-              .dashboard-page-date-range{translate: 'common.date_range.older', aria-hidden: 'true'}
-              .dashboard-page-threads
-                %dashboard_page_thread{discussion: 'discussion', ng-repeat: 'discussion in dashboardPage.dashboardDiscussions() | filter:dashboardPage.older'}
+            %dashboard_page_thread_collection{name: 'today', threads: 'dashboardPage.dashboardDiscussions', filter: 'dashboardPage.today'}
```

Comment:
I had been thinking that you would just pass 'today' into the component... and the component itself would setup a lokijs view of the right section of records. Then you don't need to use an expensive filter and the "what to display" logic is kept within the component, and you don't need to tell the component how to do it's job.

What do you think about that?